src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
@galvesribeiro I want to make sure I understand what you're suggesting.
You wrote "IMHO I think we should register all serializers that would be used, and let the grain/developer select when and which one to use."
How would the grain or developer 'select' which to use?  My understanding of the current proposal is that the selected serializer would be the first serializer that the system finds that knows how to serialize the type of data being serialized.  The grain has no control of this.  It was proposed that the order in which serializers are configured could effect this.  As more serializers are added, each wanting to be able to serialize a wide range of types of data, this seems like a recipe for chaos.

Imagine that you're using a bond serializer and a secure encrypted serializer.  You write all of your secure data using data contracts, which are not bond objects and which the encrypted serializer can serialize.  You configure the bond serializer first, then the secure encrypted serializer second.  With this configuration the bond serializer detects all bond objects while any non bond object (your own data contracts) bypass the bond serializer and uses the secure encrypted serializer.  Perfect, everything works fine.  Then someone adds support for data contracts to the bond serializer, and now all the sudden you're secure data is no longer being encrypted.

Did you have a different serializer selection process in mind?  I don't really have any issue with multiple seralizers, as long as the selection process is explicit and testable.  It should not depend on the capabilities of the serializers.  My preference for a single serializer was a means of enforcing this strict behavior, but I'm quite open to other suggestions.